### ISSUE TITLE
Changes for azurerm_monitor_aad_diagnostic_setting on every terraform apply

### ISSUE THREAD

**Issue by mlcooper** [hashicorp/terraform-provider-azurerm#16197]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.5

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

azurerm_monitor_aad_diagnostic_setting

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_aad_diagnostic_setting" "user_provisioning_logs" {
  name                           = "ManagedLogging-DML-UserProvisioningLogs"
  eventhub_authorization_rule_id = module.aad_logs_eventhub_namespace.eventhub_namespace_auth_rule_id["dml-aadlogs-diagnostic-setting"]
  eventhub_name                  = module.aad_logs_eventhub_namespace.eventhub_name["evh-dml-aadlogs-user"]
  storage_account_id             = module.storage.id["stdml${var.tenant_abbreviation}${var.environment}aaduser"]
  log_analytics_workspace_id     = local.log_analytics_workspace

  log {
    category = "ADFSSignInLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "AuditLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "ManagedIdentitySignInLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "NonInteractiveUserSignInLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "ProvisioningLogs"
    enabled  = true

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "ServicePrincipalSignInLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "SignInLogs"
    enabled  = false

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "RiskyUsers"
    enabled  = true

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  log {
    category = "UserRiskEvents"
    enabled  = true

    retention_policy {
      days    = 0
      enabled = false
    }
  }

  timeouts {}
}
```

### Debug Output/Panic Output

```shell
module.dml_tenant_azurepoc001.azurerm_monitor_aad_diagnostic_setting.user_provisioning_logs will be updated in-place
  ~ resource "azurerm_monitor_aad_diagnostic_setting" "user_provisioning_logs" {
        id                             = "/providers/Microsoft.AADIAM/diagnosticSettings/ManagedLogging-DML-UserProvisioningLogs"
        name                           = "ManagedLogging-DML-UserProvisioningLogs"
        # (4 unchanged attributes hidden)

      - log {
          - category = "NetworkAccessTrafficLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "RiskyServicePrincipals" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "ServicePrincipalRiskEvents" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (10 unchanged blocks hidden)
    }
```

### Expected Behaviour

I would expect that Terraform would not want to make any changes because I have not specified `NetworkAccessTrafficLogs`, `RiskyServicePrincipals` or `ServicePrincipalRiskEvents` in my source code. If I do add any of these, TF throws an error that these names are unexpected.

### Actual Behaviour

What actually happens (see debug output above) is that TF wants to make a change on every `terraform apply`

### Steps to Reproduce

1. Terraform apply

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/15416 may be related

**Comment by mlcooper** [comment]
Hi @stephybun, curious about why the `question` tag was applied to this issue instead of the `bug` tag.

**Comment by stephybun** [comment]
Hey @mlcooper,

Scanning through the existing issues for the monitor service I found this one https://github.com/hashicorp/terraform-provider-azurerm/issues/10388. Although it's a different diagnostic resource similar behaviours are exhibited and there is also a [workaround](https://github.com/hashicorp/terraform-provider-azurerm/issues/10388#issuecomment-1067095863) described in that issue. It also seems that this is caused by the way the API is designed (the upstream-microsoft label) which is why I categorized this as a usage question rather than a bug.

Could you give that workaround a try and let us know whether it applies to your particular scenario? That would be very helpful for us.

Thanks!

**Comment by mlcooper** [comment]
Hi @stephybun,

I gave this workaround a try, and I do not believe it will work for me. The reason is because I'm actually using the resource `azurerm_monitor_aad_diagnostic_setting` which is different than what the workaround is using.  For my `data "azurerm_monitor_diagnostic_categories"`, it does not appear as if I can target it at an existing resource, because this is different than me trying to get log categories from something like an Azure Keyvault. I have have to get the log categories from AAD which is different than a resource I create.

So I'm not sure what to point my `data` resource at to get the categories; I tried doing this, but it resulted in a `cycle` error from Terraform:

```
data "azurerm_monitor_diagnostic_categories" "signin_logs_categories" {
  resource_id = azurerm_monitor_aad_diagnostic_setting.signin_logs.id
}
```
